File: DesignPatternsDotNet/StateDesignPattern/IMobile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StateDesignPattern
{
    //Context Interface
    public interface IMobile
    {
        IState State
        {
            set;
        }
        void ChargeMobile();
        void USBConnect();
        void AdbAccess();
        IState GetCharging();
        IState GetUsb();
        IState GetAdb();
    }
}

File: DesignPatternsJava/src/main/java/com/stateDesignPattern/IMobile.java
package com.stateDesignPattern;
//Context Interface
public interface IMobile {
	public void setMobileState(IState aState);
	public void ChargeMobile();
	public void USBConnect();
	public void AdbAccess();
	public IState getCharging() ;
	public IState getUsb() ;
	public IState getAdb() ;
}
